src/button/default.scss

```diff
@@ -20,7 +20,8 @@
     @include iui-notification-marker-hover;
   }
 
-  &:focus:not(:focus-visible) {
+  // `not(:hover)` is needed for focus styles after button is clicked.
+  &:focus:not(:focus-visible):not(:hover) {
```

Comment:
I took the time to think about this some more. This is a bandaid fix which I'm afraid can cause other problems as it increases specificity even more.

One way to fix this is by changing all instances of `:focus:not(:focus-visible)` to `:focus:where(:not(:focus-visible))`. The `:where` selector doesn't add specificity so it will lower it.

The [`:where`](https://caniuse.com/mdn-css_selectors_where) selector is new but not much newer than [`:focus-visible`] itself, so I think it's fine to use here.

Reply:
With this approach, just remember to put the focus rules _before_ the hover ones so that the cascade knows which one should take preference.

If the focus and hover styles were previously grouped, a new mixin may be required.
![image](https://user-images.githubusercontent.com/9084735/144894376-78319c30-adca-4fa1-9b6d-476afd5a1bf9.png)
![image](https://user-images.githubusercontent.com/9084735/144894421-3f449df0-1308-44d4-aba2-58bea2f3c01a.png)

Reply:
oops :(